pkg/operator/clustermembercontroller/clustermembercontroller.go

```diff
@@ -89,87 +107,586 @@ func (c *ClusterMemberController) reconcileMembers(ctx context.Context, recorder
 		return fmt.Errorf("could not get list of unhealthy members: %v", err)
 	}
 	if len(unhealthyMembers) > 0 {
-		klog.V(4).Infof("unhealthy members: %v", spew.Sdump(unhealthyMembers))
+		klog.V(4).Infof("reconcile aborted found unhealthy etcd members: %v", spew.Sdump(unhealthyMembers))
 		return fmt.Errorf("unhealthy members found during reconciling members")
 	}
 
-	// etcd is healthy, decide if we need to scale
-	podToAdd, err := c.getEtcdPodToAddToMembership(ctx)
-	switch {
-	case err != nil:
-		return fmt.Errorf("could not get etcd pod: %w", err)
-	case podToAdd == nil:
-		// no more work left to do
-		return nil
-	default:
-		recorder.Eventf("FoundPodToScale", "found pod to add to etcd membership: %v", podToAdd.Name)
+	// Set desired cluster size
+	// TODO(haseeb): Always update this to react to control plane size change in cluster config?
+	if c.desiredControlPlaneSize == 0 {
+		desiredControlPlaneSize, err := ceohelpers.GetMastersReplicaCount(ctx, c.kubeClient)
```

Comment:
this is bad. We cannot do quorum read for this every single time the sync func is run, e.g. when nodes change (heartbeats).

Reply:
ic, this is only done once and then cached in the controller struct.